tests/fips/fips_setup.pm

```diff
@@ -33,76 +41,38 @@ sub run {
-        zypper_call("search -si -t pattern fips");
-
         return;
     }
 
     # FIPS_INSTALLATION is only applicable for system installaton
     die "FIPS_INSTALLATION is require to run this script for installation" if get_var("!BOOT_HDD_IMAGE");
     die "FIPS setup is only applicable for FIPS_ENABLED=1 image!" if get_var("!FIPS_ENABLED");
 
-    # If FIPS_ENV_MODE, then set ENV for some FIPS modules. It is a
-    # workaround when fips=1 kernel cmdline is not working.
-    # If FIPS_ENV_MODE does not set, global FIPS mode (fips=1 from
-    # kernel command line) will be applied
     if (get_var("FIPS_ENV_MODE")) {
         die 'FIPS kernel mode is required for this test!' if check_var('SECURITY_TEST', 'crypt_kernel');
-        zypper_call('in -t pattern fips');
+        is_alp ? trup_call("pkg install -t pattern fips") : zypper_call('in -t pattern fips');
+        $self->reboot_and_select_serial_term;
         foreach my $env ('OPENSSL_FIPS', 'OPENSSL_FORCE_FIPS_MODE', 'LIBGCRYPT_FORCE_FIPS_MODE', 'NSS_FIPS', 'GnuTLS_FORCE_FIPS_MODE') {
             assert_script_run "echo 'export $env=1' >> /etc/bash.bashrc";
         }
-
         record_info 'ENV Mode', 'FIPS environment mode (for single modules) configured!';
-    }
-    else {
-        if (is_alp) {
-            trup_call("pkg install -t pattern fips");
-        } else {
-            zypper_call('in -t pattern fips');
-        }
-        add_grub_cmdline_settings('fips=1', update_grub => 1);
-        record_info 'Kernel Mode', 'FIPS kernel mode configured!';
-    }
+    } else {
+        trup_call("pkg install crypto-policies-scripts") if is_alp;
+        zypper_call("in crypto-policies-scripts") if (is_sle('>=15-SP4') || is_jeos || is_tumbleweed);
+        zypper_call("in -t pattern fips") if is_sle('<=15-SP3');
 
-    # Try to install hmac related packages when sle >= 15-sp4
-    # Refer to poo #110707
-    if (is_sle('>=15-sp4')) {
-        my $pkg_list = {
-            'libcryptsetup12-hmac' => '2.4.3',
-            'libgnutls30-hmac' => '3.7.3',
-            'libopenssl1_1-hmac' => '1.1.1l',
-            'libgcrypt20-hmac' => '1.9.4'
-        };
-        zypper_call("in " . join(' ', keys %$pkg_list));
-        package_upgrade_check($pkg_list);
-    }
+        $self->reboot_and_select_serial_term if (is_alp || is_jeos);
 
-    if (is_alp) {
-        my $prev_console = current_console();
-        power_action('reboot', textmode => 1);
-        $self->wait_boot(bootloader_time => 300);
-        select_console($prev_console);
-    } else {
-        power_action('reboot', textmode => 1);
-    }
-    if (is_pvm) {
-        reconnect_mgmt_console;
-        $self->wait_boot(textmode => 1, ready_time => 600, bootloader_time => 300);
-    }
-    else {
-        $self->wait_boot(bootloader_time => 200);
+        if (is_alp || is_sle('>=15-SP4') || is_jeos || is_tumbleweed) {
+            assert_script_run("fips-mode-setup --enable");
+            $self->reboot_and_select_serial_term;
```

Comment:
this function might be useful here, if you don't have to reboot in ALP (do you?) but in the previous case was used it is doing some check that is not necessary, with all the installation package the reboot is needed for sure. Calling directly to `process_reboot` would be more clear.

Reply:
After fips activation we must reboot the system (this is valid for all distros), that why I've put the reboot code there.
The reboot subroutine is doing a bit more than just reboot (checking also for pvm and doing wait_boot), so I don't see the need to add another if here 🤔 (I guess you mean having something like `process_reboot if is_alp;`).
Since we always need to reboot, I did change the subroutine code though.